src/ir/item.rs

```diff
@@ -38,13 +38,46 @@ pub trait ItemCanonicalName {
 /// }
 /// ```
 ///
-/// For bar, the canonical path is `foo::BAR`, while the canonical name is just
-/// `BAR`.
+/// For bar, the canonical path is `vec!["foo", "BAR"]`, while the canonical
+/// name is just `"BAR"`.
 pub trait ItemCanonicalPath {
     /// Get the canonical path for this item.
     fn canonical_path(&self, ctx: &BindgenContext) -> Vec<String>;
 }
 
+/// A trait for iterating over an item and its parents and up its ancestor chain
+/// up to (but not including) the implicit root module.
+pub trait ItemAncestors {
+    /// Get an iterable over this item's ancestors.
+    fn ancestors<'a, 'b, 'c>(&'a self,
```

Comment:
I think you can write this as:

```rust
fn ancestors<'a, 'ctx>(&self, ctx: &'a BindgenContext<'ctx>) -> ItemAncestorsIter<'a, 'ctx>
```

Or it might be worth to just use `&'a BindgenContext<'a>`, and use a single lifetime if you can.